minio-client/deployment/src/main/java/io/quarkiverse/minio/client/deployment/MinioClientProcessor.java

```diff
@@ -73,21 +68,37 @@ void produce(MiniosBuildTimeConfiguration miniosBuildTimeConfiguration,
                 .setDefaultScope(DotNames.SINGLETON).build());
         for (Map.Entry<String, MinioBuildTimeConfiguration> entry : miniosBuildTimeConfiguration.getMinioClients().entrySet()) {
             var minioClientName = entry.getKey();
-            SyntheticBeanBuildItem.ExtendedBeanConfigurator configurator = SyntheticBeanBuildItem
-                    .configure(MinioClient.class)
-                    .scope(Singleton.class)
-                    .setRuntimeInit()
-                    .unremovable()
-                    // Pass runtime configuration to ensure initialization order
-                    .supplier(minioRecorder.minioClientSupplier(minioClientName, miniosRuntimeConfiguration));
-            if (MiniosBuildTimeConfiguration.isDefault(minioClientName)) {
-                configurator.addQualifier(DotNames.DEFAULT);
-            } else {
-                configurator.addQualifier().annotation(DotNames.NAMED).addValue("value", minioClientName).done();
-                configurator.addQualifier().annotation(MinioQualifier.class)
-                        .addValue("value", minioClientName).done();
-            }
-            syntheticBeanBuildItemBuildProducer.produce(configurator.done());
+            syntheticBeanBuildItemBuildProducer.produce(
```

Comment:
I also think async should be opt-in via configuration, in this case we could only produce the necessary beans.

What do you think ?

Reply:
Hmm, I understand where you are coming from. At the same time, I would prefer it to just work(TM) as the async client is a default feature offered by the Minio client library. I'd prefer not to hide default functionality behind a flag.

Reply:
We opt in everything but that would break applications already using the extension.

But I don't want to create useless objects

Reply:
I'm not sure I understand how adding an extra bean would break existing applications. Anyway, if we don't mark them as unremoveable they are pruned by Quarkus when unused. Or did I misunderstand something there?

Reply:
Automatically adding the async client won't break anything, but that would produce unused bean. As of now the producer are marked unremovable. You can try to make them removable but I think I has issues without it.

Reply:
Only the produced `MinioClient` beans are marked unremovable, not the producer.

I've looked up the pertinent documentation regarding removed beans:
- [CDI Reference](https://quarkus.io/guides/cdi-reference#whats-removed)
- [Blog: Unused Beans](https://quarkus.io/blog/unused-beans/)

According to those pages the client beans wouldn't be removed as long as they are injected somewhere. The `MinioClients` class I would leave unremovable. But I think even that isn't necessary.

Reply:
Yes that's the one I was speaking about, nevermind, let's say that one synthetic bean is not awfull